### ISSUE TITLE
Pay to increase reliability in career

### ISSUE THREAD

**Issue by TheSpiritOfMan** [zer0Kerbal/OhScrap#6]
This mod brings great excitement to the career in KSP. However, since all newly unlocked parts are almost guaranteed to fail the first time you try them, you need lots of actual tests. That is accompanied with lots of loading, scene changes and wasted time.

**I would love to see a feature where you can pay a fee in career to up the reliability for a specific part.**
That would allow to skip the tedious first few manual tests, but still keep the excitement of failure.

For balance reasons I would like to propose exponentially increasing testing costs. E.g.:

```
Generation 0 -> 1: 0.5 x unlock cost
Generation 1 -> 2: 1 x unlock cost
Generation 2 -> 3: 2 x unlock cost
Generation 3 -> 4: 4 x unlock cost
...
```

Allowing you to:

1. Pay a little to skip the first initial (almost inevitable failure)
2. Pay excessive amounts of money to make sure absolutely crucial missions are safer

**Comment by chrbkr** [comment]
I wouldn't use unlock cost, but part cost as a basis. Unlock cost can be 0 or very, very expensive. Part cost is what you pay now. The only thing you save with this feature is time.

Note that as of now, there's a difference between tested and next generation. For testing, it's enough to roll out and roll back. That's cumbersome, because of loading times, scene change etc. On the other hand, if you do a static fire test on the pad, then refuel and launch, engine and tanks are still not tested.
While I guess it's hard to tell what "test conditions" are, I think it should be something like this: You get a button in the VAB that allows you to test a part/the entire vessel. Testing costs an extra fee, say, 5% of the part cost. Clicking the button then basically rolls a dice: with the usual probability for untested parts, the parts can fail. If that's the case, the failed parts are "replaced", i.e. their cost is added again. Now you have vessel with all parts tested.

Advancing to the next generation is a different thing. Exponential cost increase would be a big difference from current gameplay, as advancing to the next generation is how basically free. Maybe that would also make sense to be combined with failures: as long as nothing goes wrong, advancement remains free. But a failure would need some redesign, and that becomes more expensive with each generation.
While that would work with engines for first and second stages as they are used only for a short time, I don't know how to handle satellite and probe parts, though.

**Comment by TheSpiritOfMan** [comment]
Obviously, I should have suggested using cost, agreed. 
I do like having varying costs for "virtual testing" in accordance with the actual failure probabilities. 

You are right, I did not think about the difference between parts being tested or being the next generation. Paying for having parts tested, as you suggested above would achieve what I would like to see: 
Paying a little to move from _almost guaranteed to fail_ to _might actually work_ without compromising the gameplay of improving reliability by building subsequent generations.
Since having the option to pay to advance to the highest reliability would eliminate what makes this mod so interesting: With it in career one has to choose whether to use tried and tested, but less advanced or more advanced, but very unreliable parts.

**Comment by chrbkr** [comment]
Unlike other mods, "advanced" doesn't give any advantage other than being more reliable. Maybe the authors can elaborate on what's the difference between a tested and a next gen part in terms of failure probability?

On a side note: since this mod, ScrapYard and KCT work well together, maybe the testing part could be moved to KCT, as testing would also take some time. Now that I think of it, developing the next generation would also take time...

**Comment by TheSpiritOfMan** [comment]
While that seems sensible, moving it to KCT would lead to a lot more switching between VAB and Space Center and I think those are unnecessary loading times. There's probably a reason KCT demands research time to unlock new tech tree notes in bulk. I think introducing a testing time for each part would maybe be a bit much.

**Comment by chrbkr** [comment]
Yes that would be too much. I just meant an option to "test it before launch, so just add some more time to build or rollout time"

**Comment by zer0Kerbal** [comment]
#### see also #31 (Generational Improvements via Tech Tree)

if #31 is possible, so is this. just have to 
- [ ] would have to be and optional enhancement, default off
- [ ] limit to say three generations purchasable
- [ ] should have to be via each part.
- [ ] could also be tied into game difficulty mode
- [ ] Generational costs, benefits:
  - [ ] Generation +1: 0.5 x unlock cost
  - [ ] - Generation +2: 1 x unlock cost
  - [ ] - Generation +3: 2 x unlock cost
  - [ ] - Generation +4: 4 x unlock cost
